scripts/actor/actor-wfrp4e.js

```diff
@@ -3271,52 +3293,151 @@ class ActorWfrp4e extends Actor {
+   * However, I find deleting the ones you don't want to be much easier than trying to pick and 
+   * choose the ones you do want.
+   *
+   * @param {Object} careerData     Career type Item to be used for advancement.
+   * 
+   * TODO Refactor for embedded entity along with the helper functions
+   */
+  async _advanceNPC(careerData) 
+  {
+    let updateObj = {};
+    let advancesNeeded = careerData.level.value * 5; // Tier 1 needs 5, 2 needs 10, 3 needs 15, 4 needs 20 in all characteristics and skills
 
+    // Update all necessary characteristics to the advancesNeeded
+    for (let advChar of careerData.characteristics)
+      if (this.data.data.characteristics[advChar].advances < 5 * careerData.level.value)
+        updateObj[`data.characteristics.${advChar}.advances`] = 5 * careerData.level.value;
 
-    _replaceData(formula) {
-      let dataRgx = new RegExp(/@([a-z.0-9]+)/gi);
-      return formula.replace(dataRgx, (match, term) => {
-        let value = getProperty(this.data, term);
-        return value ? String(value).trim() : "0";
-      });
+    // Advance all skills in the career
+    for (let skill of careerData.skills)
+      await this._advanceSkill(skill, advancesNeeded);
+
+    // Add all talents in the career
+    for (let talent of careerData.talents)
+      await this._advanceTalent(talent);
+
+    this.update(updateObj);
+  }
+
+
+  _replaceData(formula) {
+    let dataRgx = new RegExp(/@([a-z.0-9]+)/gi);
+    return formula.replace(dataRgx, (match, term) => {
+      let value = getProperty(this.data, term);
+      return value ? String(value).trim() : "0";
+    });
+  }
+
+  /**
+   * Use a fortune point from the actor to reroll or add sl to a roll
+   * @param {Object} message 
+   * @param {String} type (reroll, addSL)
+   */
+  useFortuneOnRoll(message, type)
+  {
+    if(this.data.data.status.fortune.value > 0)
+    {
+      let data = message.data.flags.data;
+      let html = `<h3 class="center"><b>${game.i18n.localize("FORTUNE.Use")}</b></h3>`;
+      //First we send a message to the chat
+      if(type=="reroll")
+        html += `${game.i18n.format("FORTUNE.UsageRerollText",{character:'<b>'+this.name+'</b>'})}<br>`;
+      else
+        html += `${game.i18n.format("FORTUNE.UsageAddSLText",{character:'<b>'+this.name+'</b>'})}<br>`;
+
+      html += `<b>${game.i18n.localize("FORTUNE.PointsRemaining")} </b>${this.data.data.status.fortune.value-1}`;
+      ChatMessage.create(WFRP_Utility.chatDataSetup(html));
+
+      let cardOptions = this.preparePostRollAction(message);
+      //Then we do the actual fortune action
+      if(type=="reroll")
+      {
+        cardOptions.fortuneUsedReroll = true;
+        ActorWfrp4e[data.postData.postFunction](data.preData,cardOptions);
+        //We also set fortuneUsedAddSL to force the player to use it on the new roll
+        message.update({
+          "flags.data.fortuneUsedReroll" : true,
+          "flags.data.fortuneUsedAddSL" : true
+        });
```

Comment:
Why are both true if type == "reroll"?

Reply:
As said in the comment, it is to prevent the player from adding 1 SL on a card to a roll who has as been rerolled. It prevent the contextmenu from displaying the addsl option. Instead, he can add SL on the newer rerolled card.